design/17-driver-comm/01-inline-ring.md

```
@@ -112,32 +119,63 @@ fn slot_ptr(base: *mut u8, header: &ShmHeader, idx: u64) -> *mut u8 {
 driver 起動時、Bridge 側で shm を確保するまでの流れ:
 
 1. **driver 起動** → events.yaml をパース
-2. **driver 側で max_payload_size を計算**:
-   - 全 events の `bytes.max_length` を収集
-   - msgpack encode 後の最大長 + 固定オーバーヘッド（type 文字列・key 名・msgpack タグ等）
-   - 結果を `max_payload_size: u32` として保持
-3. **driver → Bridge** に `request_ring(max_payload_size)` を送信（control channel 経由）
+2. **driver 側で `max_payload_size` と必要 `slot_size` を計算**（algorithm は実装間で deterministic に揃える）:
+   - **`tier: inline`**（または default）の event のみ対象
+   - **各 event について msgpack worst-case payload サイズを算出**（同一 event 内の可変長フィールドが**同時に**最大値を取るケースを採る = 過小見積もり防止）。算出規約は次の通り:
+     | 要素 | worst-case サイズ |
```

markdownlint-cli2 (0.22.1)

[warning] Tables should be surrounded by blank lines

(MD058, blanks-around-tables)